cmd/kubelet/app/server.go

```diff
@@ -126,7 +126,7 @@ func UnsecuredKubeletDeps(s *options.KubeletServer) (*kubelet.KubeletDeps, error
 
 	var dockerClient dockertools.DockerInterface
 	if s.ContainerRuntime == "docker" {
-		dockerClient = dockertools.CreateDockerClientOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
+		dockerClient = dockertools.ConnectToDockerOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
```

Comment:
Are we sure we want to connect to the docker client this early? IMO the decision comes down to whether we typically want the docker client to already be connected to the daemon when we inject the client into the Kubelet, or if at this early a stage we just want the client to be set up so it can try connecting later. The former fails faster if it can't connect, the latter is lighter weight as far as building dependencies to inject. What do you think?